File: Code coach/Hard/Security/Security.java
import java.util.Scanner;

public class Security {

	public static void main(String[] args) {
		Scanner input = new Scanner(System.in);

		String diagram = input.nextLine();
		String[] diagramSplitted = diagram.split("\\$");

		boolean isThiefOnLeft = true;
		int thiefPos = -1;
		int guardPos = -1;

		int idx1 = 0;
		for (String part : diagramSplitted) {
			if (part.contains("T")) {
				int idx2 = 0;
				for (char c : part.toCharArray()) {
					if (Character.toUpperCase(c) == 'G') {
						if (!isThiefOnLeft && guardPos != -1) {
							continue;
						} else {
							guardPos = idx2;
						}
					} else if (Character.toUpperCase(c) == 'T') {
						thiefPos = idx2;
					}
				}
			} else if (idx1 == 0) {
				isThiefOnLeft = false;
				continue;
			}
		}

		boolean isLeftRobbed = (guardPos == -1 || guardPos < thiefPos) && isThiefOnLeft;
		boolean isRightRobbed = (guardPos == -1 || guardPos > thiefPos) && !isThiefOnLeft;

		if (isLeftRobbed || isRightRobbed) {
			System.out.print("ALARM");
		} else {
			System.out.print("quiet");
		}

		input.close();
	}

}

File: Code coach/Hard/Security/security.py
diagram = input()
diagramSplitted = diagram.split('$')

isThiefOnLeft = True
thiefPos = None
guardPos = None

for idx1, part in enumerate(diagramSplitted):
	if 'T' in part:
		for idx2, c in enumerate(part):
			if c.upper() == 'G':
				if not isThiefOnLeft and guardPos is not None:
					continue
				else:
					guardPos = idx2
			elif c.upper() == 'T':
				thiefPos = idx2
	elif idx1 == 0:
		# if the field is not in the first array (left), then we know that it is in the right
		isThiefOnLeft = False
		continue

isLeftRobbed = (guardPos is None or guardPos < thiefPos) and isThiefOnLeft
isRightRobbed = (guardPos is None or guardPos > thiefPos) and not isThiefOnLeft

if isLeftRobbed or isRightRobbed:
	print("ALARM")
else:
	print("quiet")
